charts/ingress-nginx/templates/controller-poddisruptionbudget.yaml

```diff
@@ -1,4 +1,10 @@
-{{- if or (and .Values.controller.autoscaling.enabled (gt (.Values.controller.autoscaling.minReplicas | int) 1)) (and (not .Values.controller.autoscaling.enabled) (gt (.Values.controller.replicaCount | int) 1)) }}
+{{- $replicas := .Values.controller.replicaCount }}
+{{- if .Values.controller.autoscaling.enabled }}
+{{- $replicas = .Values.controller.autoscaling.minReplicas }}
+{{- else if .Values.controller.keda.enabled }}
+{{- $replicas = .Values.controller.keda.minReplicas }}
```

Comment:
Don't we need to add some docs for this?

What if the user enable both by mistake?
Can we try to use the OR instead or that does not work?


/hold for clarification

Reply:
Hm, yeah, I get your point. HPA and KEDA being mutually exclusive always was a problem and looking at the particular resource templates, we are not even preventing that in there.

So right now what you see above is pretty close the best we can get without making the conditions too complex and unreadable.

I created an issue for improving and aligning that everywhere. But for now, I'd just go on and merge this PR. Aligning all these conditions is out of scope here.

Reply:
That's the issue I created: https://github.com/kubernetes/ingress-nginx/issues/10814